gnovm/tests/files/zrealm_crossrealm11.gno

```diff
@@ -0,0 +1,152 @@
+		"user1.gno", "gno.land/r/crossrealm_test", "gno.land/r/demo/tests",
+	} {
+		addr := std.DerivePkgAddr(caller)
+		callersByAddr[addr] = caller
+	}
+
+	baseCallStack := "user1.gno -> r/crossrealm_test.main()"
+	tests := []struct {
+		callStackAdd string
+		callerFn     func() std.Address
+	}{
+		{
+			callStackAdd: "",
+			callerFn:     std.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> r/crossrealm_test.getRealmCaller()",
+			callerFn:     getRealmCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests",
+			callerFn:     ptests.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests -> p/demo/tests/subtests",
+			callerFn:     ptests.GetSubtestsCaller,
+		},
+		{
+			callStackAdd: " -> r/demo/tests",
+			callerFn:     rtests.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> r/demo/tests -> r/demo/tests/subtests",
+			callerFn:     rtests.GetSubtestsCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests -> r/demo/tests",
+			callerFn:     ptests.GetRealmCaller,
+		},
+	}
+	for i, tt := range tests {
+		callStack := ufmt.Sprintf("%s%s -> std.GetRealmCaller()", baseCallStack, tt.callStackAdd)
+		println(ufmt.Sprintf("STACK CASE %d:\n%s", i, callStack))
+		println(ufmt.Sprintf("\tcaller is %s", callersByAddr[tt.callerFn()]))
+		Exec(func() {
+			println(ufmt.Sprintf("wrapped in r/crossrealm_test.Exec:\n\tcaller is %s", callersByAddr[tt.callerFn()]))
+		})
+		rtests.Exec(func() {
+			println(ufmt.Sprintf("wrapped in r/demo/tests.Exec:\n\tcaller is %s", callersByAddr[tt.callerFn()]))
+		})
+		ptests.Exec(func() {
+			println(ufmt.Sprintf("wrapped p/demo/tests.Exec:\n\tcaller is %s", callersByAddr[tt.callerFn()]))
+		})
+		println()
+	}
+}
+
+// Output:
+// STACK CASE 0:
+// user1.gno -> r/crossrealm_test.main() -> std.GetRealmCaller()
+// 	caller is user1.gno
+// wrapped in r/crossrealm_test.Exec:
+// 	caller is user1.gno
+// wrapped in r/demo/tests.Exec:
+// 	caller is gno.land/r/demo/tests
+// wrapped p/demo/tests.Exec:
+// 	caller is user1.gno
+//
+// STACK CASE 1:
+// user1.gno -> r/crossrealm_test.main() -> r/crossrealm_test.getRealmCaller() -> std.GetRealmCaller()
+// 	caller is user1.gno
+// wrapped in r/crossrealm_test.Exec:
+// 	caller is user1.gno
+// wrapped in r/demo/tests.Exec:
+// 	caller is gno.land/r/demo/tests
+// wrapped p/demo/tests.Exec:
+// 	caller is user1.gno
+//
+// STACK CASE 2:
+// user1.gno -> r/crossrealm_test.main() -> p/demo/tests -> std.GetRealmCaller()
+// 	caller is user1.gno
+// wrapped in r/crossrealm_test.Exec:
+// 	caller is user1.gno
+// wrapped in r/demo/tests.Exec:
+// 	caller is gno.land/r/demo/tests
+// wrapped p/demo/tests.Exec:
+// 	caller is user1.gno
+//
+// STACK CASE 3:
+// user1.gno -> r/crossrealm_test.main() -> p/demo/tests -> p/demo/tests/subtests -> std.GetRealmCaller()
+// 	caller is user1.gno
+// wrapped in r/crossrealm_test.Exec:
+// 	caller is user1.gno
+// wrapped in r/demo/tests.Exec:
+// 	caller is gno.land/r/demo/tests
+// wrapped p/demo/tests.Exec:
+// 	caller is user1.gno
```

Comment:
Wouldn't it be like this instead?

```suggestion
// user1.gno -> r/crossrealm_test.main() -> p/demo/tests -> p/demo/tests/subtests -> std.GetRealmCaller()
// 	caller is crossrealm_test
// wrapped in r/crossrealm_test.Exec:
// 	caller is crossrealm_test
// wrapped in r/demo/tests.Exec:
// 	caller is gno.land/r/demo/tests
// wrapped p/demo/tests.Exec:
// 	caller is crossrealm_test
```

Reply:
Why that? 

>  user1.gno -> r/crossrealm_test.main() -> p/demo/tests -> p/demo/tests/subtests -> std.GetRealmCaller()

The caller is user1.gno because there is just 1 realm

>  user1.gno -> r/crossrealm_test.main() -> p/demo/tests -> p/demo/tests/subtests -> r/crossrealm_test.Exec(std.GetRealmCaller())

This one is a bit more tricky, because the realm is calling itself, So technically, the stack have twice the Realm, but in the code, we use a hashmap to only keep 1.

But same as previous, there is only 1 realm called.

>  user1.gno -> r/crossrealm_test.main() -> p/demo/tests -> p/demo/tests/subtests -> p/demo/tests.Exec(std.GetRealmCaller())

Only packages are called, GetRealmCaller return the user.



Maybe this will help understand, the future of foo20.Transfer is to be like:

```go
func Transfer(to users.AddressOrName, amount uint64) {
	caller := std.GetRealmCaller()
	foo.Transfer(caller, to.Resolve(), amount)
}
```

So when there is 1 realm, we return the user, if more, we return the realm

Reply:
Apologies for the confusion. I misinterpreted `r/demo/tests/subtests` as `p/demo/tests/subtests`. Thank you for correcting me.